src/main/java/chess/domain/square/Square.java

```diff
@@ -0,0 +1,99 @@
+package chess.domain.square;
+
+import chess.domain.piece.PieceColor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public record Square(File file, Rank rank) {
+
+    private static final int ONE_STEP = 1;
+    private static final int TWO_STEP = 2;
+
+    public boolean isStraight(final Square target) {
+        return file.isSameFile(target.file) || rank.isSameRank(target.rank());
+    }
+
+    public boolean isDiagonal(final Square target) {
+        return file.calculateDistance(target.file) == rank.calculateDistance(target.rank);
+    }
+
+    public boolean isWithinOneStep(final Square target) {
+        return file.calculateDistance(target.file) <= ONE_STEP &&
+                rank.calculateDistance(target.rank) <= ONE_STEP;
+    }
+
+    public boolean isStraightAndDiagonal(final Square target) {
+        if (file.calculateDistance(target.file) == ONE_STEP
+                && rank.calculateDistance(target.rank) == TWO_STEP) {
+            return true;
+        }
+        return file.calculateDistance(target.file) == TWO_STEP
+                && rank.calculateDistance(target.rank) == ONE_STEP;
+    }
+
+    public boolean isNotBackward(final Square target, final PieceColor color) {
+        if (color.isBlack()) {
+            return rank.calculateDirection(target.rank) > 0;
+        }
+        return rank.calculateDirection(target.rank) < 0;
+    }
+
+    public boolean isOnlyForward(final Square target) {
+        if (isFirstMove()) {
+            return rank.calculateDistance(target.rank) <= TWO_STEP
+                    && file.isSameFile(target.file);
+        }
+        return rank.calculateDistance(target.rank) == ONE_STEP
+                && file.isSameFile(target.file);
+    }
+
+    private boolean isFirstMove() {
+        return rank.isSameRank(Rank.TWO) || rank.isSameRank(Rank.SEVEN);
+    }
+
+    public boolean isAttack(final Square target) {
+        if (canAttack(target)) {
+            return isDiagonal(target)
+                    && file.calculateDistance(target.file) == 1;
```

Comment:
```suggestion
            return rank.calculateDistance(target.rank) <= TWO_STEP && file.isSameFile(target.file);
        }
        return rank.calculateDistance(target.rank) == ONE_STEP && file.isSameFile(target.file);
    }

    private boolean isFirstMove() {
        return rank.isSameRank(Rank.TWO) || rank.isSameRank(Rank.SEVEN);
    }

    public boolean isAttack(final Square target) {
        if (canAttack(target)) {
            return isDiagonal(target) && file.calculateDistance(target.file) == 1;
```
내려간 이유가 있을까요? 제기억에 우테코에서 제공된 컨벤션으로는 이정도 길이면 인텔리제이 리펙터링에서 안내려가는 길이로 알고 있어서요. (아니라면 넘어가주세요 😄 )

Reply:
제가 보기에 가독성이 안 좋은 경우 한 칸 내렸습니다.😸
컨벤션에 맞게 다시 한 줄로 고쳤습니다! ㅎㅎㅎ